### ISSUE TITLE
Gzip and cache headers missing for some resources

### ISSUE THREAD

**Issue by digitarald** [mozilla/testpilot#1916]
Time to Meaningful Paint is about 10sec on DSL, which probably causes higher bounce rates than necessary. I talked to @ckprice about performance before about tracking this; so I'll just start with small nits:

Missing gzip: https://www.webpagetest.org/result/161214_EP_4XC6/1/performance_optimization/#compress_text
Missing caching: https://www.webpagetest.org/result/161214_EP_4XC6/1/performance_optimization/#cache_static_content

**Comment by digitarald** [comment]
The 10sec are painfully highlighted in this [slow-mo video for DSL](https://www.webpagetest.org/video/view.php?id=161214_747e61f24c41c57df5a14118ee2c3e57f6d1d951&embed=1&width=520&height=448)

**Comment by ckprice** [comment]
Chatted a bit with Harold. Taking this for first steps.

**Comment by lmorchard** [comment]
Probably some checkboxes need checking in our stage & prod CloudFront distributions managed by ops? Would be interesting to see if this is an issue on testpilot.dev.mozaws.net, since that's a CloudFront distro we manage on the dev team

**Comment by ckprice** [comment]
@relud is this something you can configure?

**Comment by relud** [comment]
gzip is probably a flag on my end somewhere, but caching should be set by bin/deploy.sh

**Comment by lmorchard** [comment]
TL;DR: We can knock off some of those problems WebPagetest reports, but we won't get to 100%

* We are indeed missing a `--cache-control "max-age=${MAX_AGE}"` on [deploying `*.svg` assets](https://github.com/mozilla/testpilot/blob/master/bin/deploy.sh#L108), so we can add that and resolve most of the issues.

* The [`experiments.json` resource](https://testpilot.firefox.com/api/experiments.json) *should* have a short cache time - the data changes whenever we deploy, but the URL can't change because 3rd parties and add-ons depend on it.

* We don't really want to use long-expiration on [the Telemetry counts](https://analysis-output.telemetry.mozilla.org/testpilot/data/installation-counts/latest.json) as that's a frequently-updated data source - and it's not under the control of TxP deployment anyway

**Comment by relud** [comment]
found where to put the gzip flag. if mozilla-services/cloudops-deployment#308 is acceptable then i will deploy that to stage, and then to prod on tuesday morning or something?

**Comment by lmorchard** [comment]
FWIW, I think that's a private repo I can't see

**Comment by relud** [comment]
ah. I added `Compress: true` to a cloudformation resource that configures cloudfront. I'm waiting on review from someone on my team.

**Comment by relud** [comment]
that said, I can deploy this to stage without approval, so I've initiated that

**Comment by ghost** [comment]
related to #1349 ?

**Comment by ckprice** [comment]
Just ran [the test again](https://www.webpagetest.org/result/170224_67_10N4/1/performance_optimization/). It's still yelling at us for Gzip, but caching seems to be fixed by #1923.

https://github.com/mozilla/testpilot/issues/1916#issuecomment-267700250 noted a `Compress: true` changed in CloudFront. Wondering if we need anything else? [Docs suggest](https://aws.amazon.com/blogs/aws/new-gzip-compression-support-for-amazon-cloudfront/) other possible configs (although could be the same thing). (/cc @relud)

**Comment by relud** [comment]
This was just deployed to prod. it had been in stage for a long time, but hadn't been scheduled for deploy to prod.

**Comment by ghost** [comment]
Thanks!

**Comment by ckprice** [comment]
Still getting an "F" on production, but not stage (maybe it takes longer than an hour for things to take effect?). Re-opening and will check again tomorrow.

Prod: https://www.webpagetest.org/result/170307_X1_1EZB/1/performance_optimization/
Stage: https://www.webpagetest.org/result/170307_B3_1EX9/1/performance_optimization/

**Comment by ghost** [comment]
@relud ping ^

**Comment by ckprice** [comment]
> Re-opening and will check again tomorrow.

Just following up that we're still getting a bad grade on prod: https://www.webpagetest.org/result/170310_26_16Z2/1/performance_optimization/

**Comment by relud** [comment]
I'm currently investigating this

**Comment by relud** [comment]
Fixed. The setting needed a cache invalidation in order to activate. https://www.webpagetest.org/result/170310_K0_18HZ/1/performance_optimization/
